aspnet/fundamentals/file-providers/sample/src/FileProviderSample/Controllers/HomeController.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.FileProviders;
+
+namespace FileProviderSample.Controllers
+{
+    public class HomeController : Controller
+    {
+        private readonly IFileProvider _fileProvider;
+        private static string _dateFileLastChanged = "Never";
+
+        public HomeController(IFileProvider fileProvider)
+        {
+            _fileProvider = fileProvider;
+        }
+
+        public IActionResult Index()
+        {
+            var contents = _fileProvider.GetDirectoryContents("");
+            return View(contents);
+        }
+
+        public IActionResult Watch()
```

Comment:
I created a console sample. It's trying to continuously monitor a file, but I'm not confident in my approach. How would you change/update it? I don't like the recursive nature of what I have, and it also fires off twice on the first change, currently.